apps/blockchain/lib/blockchain/p2p/server.ex

```diff
@@ -71,6 +71,8 @@ defmodule Blockchain.P2P.Server do
         :gen_tcp.send(socket, "unknown type")
       {:error, :invalid} ->
         :gen_tcp.send(socket, "invalid json")
+      {:error, reason} ->
+        Logger.info fn -> reason end
```

Comment:
you could do `Logger.info "mission accomplished"` instead of a function right?! I am not sure why you used a function.

Reply:
Yes totally but from the doc: https://hexdocs.pm/logger/master/Logger.html

> The Logger macros also accept messages as strings, but keep in mind that strings are always evaluated regardless of log-level. As such, it is recommended to use a function whenever the message is expensive to compute.

So even if my messages are not expensive to compute or I'm logging a variable, I took the habit to use function everywhere

Reply:
Well I didn't read that one! I think are two now, always use functions ❤️